andela/travel_tool_back
# #161044557 Edit checklist items

#### What does this PR do?
This PR enables Travel Admins and Super Admins to update a checklist item

#### Description of Task to be completed?
A Travel Admin/Super Admin for a location should be able to edit a previously set travel checklist and the corresponding output of each checklist item

#### How should this be manually tested?
- Clone the repo - `git clone https://github.com/andela/travel_tool_back.git`
- Change into the project directory -` cd travel_tool_back`
- Install project dependencies using `yarn install`
- Run database migrations using `yarn run db:rollmigrate`
- Start the server using `yarn start:dev`
- Clone the repo - `git clone https://github.com/andela/travel_tool_front.git`
- Change into the project directory -` cd travel_tool_front`
- Install project dependencies using `yarn install`
- Start the frontend using `yarn start`
- Navigate to [travela](http://travela-staging.andela.com/) in your browser and login
- Open the developer tools, locate applications tab, copy `jwt-token` in the cookies
- Check `.env.example` and set up your `.env` appropriately

- In`Postman`, use the `jwt-token` to send a request by adding a `Header`, key `Authorization` and value the `token`.
- In Postman, in the `x-www-form-urlencoded` body options;
    - Send a PUT request to `/api/v1/checklist/:id`. Using the following request body

Note: /:id (can be 1-8)
```
{
	"name": "Trip Budget",
	"requiresFiles": true,
	"resources": [
	{
	    "link": "linktoreceipt.co",
	    "label": "budget"
	}
  ]
}
```
- You should be able to see updated checklist item
- The response should include a message and a success flag

#### Any background context you want to provide?

#### What are the relevant pivotal tracker stories?
[#161044557 ](https://www.pivotaltracker.com/story/show/161044557 )

#### Screenshots (if appropriate)
<img width="1033" alt="screen shot 2018-10-11 at 6 08 18 pm" src="https://user-images.githubusercontent.com/32720508/46823716-d7861300-cd86-11e8-95ca-d2529dc43337.png">

#### Questions:
